controllers/vspherevm_controller.go

```diff
@@ -399,6 +399,8 @@ func (r vmReconciler) reconcileNormal(ctx *context.VMContext) (reconcile.Result,
 	// Get or create the VM.
 	vm, err := r.VMService.ReconcileVM(ctx)
 	if err != nil {
+		// reset VSphereVM.Status to false in case it's used to get ready but currently there's an error
+		ctx.VSphereVM.Status.Ready = false
```

Comment:
@chrischdi Thanks for your review! Yes, setting `Ready` back to `false` is newly introduced in this PR, as for instance, if we mark the condition `NotFoundByBIOSUUIDReason` as `false` after a VSphereVM is created successfully but not found for now, it looks not reasonable to have a `Ready` VSphereVM with a failed condition in status.

And for other errors, I think there's not too much change, since most of the errors are happened during VSphereVM creation, and at that time `Ready` is `false` by default.

Waiting for more thoughts about behaviors of MHC.

Reply:
Have to read through a few core CAPI controllers. I think in general the idea is that status.ready stays ready once it has been ready. Everything else is then covered by conditions: https://cluster-api.sigs.k8s.io/developer/providers/machine-infrastructure.html

Reply:
In general I'm a bit hesitent setting this field without understand the full implications of it. (we also ddin't set it to false when we were still treating this as a terminal failure)

Reply:
I'd also propose to drop this part.

Reply:
Did a bit more research. Looks like this would only set ready to false on the VSphereVM. It wouldn't propagate to VSphereMachine and it wouldn't affect MHC. But that also raises the question if it makes sense that the VSphereVM has .status.ready = false while VSphereMachine has ready = true etc.

I think for this PR we should just set the condition. We can consider making this change, but this would require extensive research how we want to propagate the ready field across resources up the hierarchy and how core CAPI would deal with it (the contract only documents the transition to ready to signal a Machine was provisioned: "Set status.ready to true"
(https://cluster-api.sigs.k8s.io/developer/providers/machine-infrastructure.html).

So by setting it to false we would definitely diverge from the specification (at least on the VSphereMachine level, VSphereVM is not specified as it's CAPV specific). 

But I think overall conditions are the main thing that are used nowadys for this sort of information.

Reply:
@chrischdi @sbueringer 
Thanks for the detailed review and extensive investigations on the PR.

That's make sense, it seems more appropriate to only set the condition in this PR without stepping into the status.Ready propagation for different CDs which may need further investigations.